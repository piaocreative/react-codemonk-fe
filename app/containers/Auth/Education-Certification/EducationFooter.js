import React from 'react';
import { FormattedMessage } from 'react-intl';
import { LinkButtonMod, Button } from 'components';
import StorageService from 'utils/StorageService';
import containerMessage from 'containers/messages';
import { TALENT_STEP_3_URL } from 'containers/Auth/constants';
import { handleBackButton } from 'containers/Auth/utils';
import { propTypes } from 'containers/proptypes';

const EducationFooter = ({ history, formValid, handleSaveForLater }) => (
  <div className="d-flex flex-column align-items-center flex-md-row justify-content-md-end my-5">
    <LinkButtonMod
      color="link"
      onClick={e => {
        handleSaveForLater(e);
      }}
    >
      <FormattedMessage {...containerMessage.skipButton} />
    </LinkButtonMod>
    <Button
      disabled={!formValid}
      className="btn-primary mt-3 mt-md-0 ms-md-3"
      onClick={e => {
        StorageService.set('signupStep', 3, { hash: true });
        handleBackButton(e, history, TALENT_STEP_3_URL);
      }}
    >
      <FormattedMessage {...containerMessage.continueButton} />
    </Button>
  </div>
);

export default EducationFooter;

EducationFooter.propTypes = propTypes;
