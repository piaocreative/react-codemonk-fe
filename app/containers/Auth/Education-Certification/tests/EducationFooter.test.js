import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import EducationFooter from '../EducationFooter';

const props = {
  educationDetails: [
    {
      degreeLevel: 'Masterâ€™s or Higher',
      degreeTitle: 'Master in Computer Application',
      collegeName: 'IETE, New Delhi',
      country: 'India',
      startDate: '14/06/2019',
      endDate: '14/06/2020',
    },
  ],
  certificateDetails: [
    {
      name: 'AWS Solution Architect',
      dateObtained: '30/08/2019',
      issuedBy: 'Amazon',
      certificateId: 'ABC123',
    },
  ],
  dispatch: jest.fn(),
  handleSubmit: jest.fn(),

  invalid: false,
  loading: false,
  responseSuccess: false,
  responseError: false,
  onReset: jest.fn(),
  history: { push: jest.fn(), replace: jest.fn() },

  onChangeEducationDetails: jest.fn(),
  onChangeCertificateDetails: jest.fn(),
  onSaveForLater: jest.fn(),
  onSubmitEducationForm: jest.fn(),

  onAddEducationForm: jest.fn(),
  onAddCertificateForm: jest.fn(),
  onDeleteEducationForm: jest.fn(),
  onDeleteCertificateForm: jest.fn(),
};

describe('<EducationFooter />', () => {
  const renderer = new ShallowRenderer();
  it('should render and match the snapshot', () => {
    renderer.render(<EducationFooter {...props} />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
