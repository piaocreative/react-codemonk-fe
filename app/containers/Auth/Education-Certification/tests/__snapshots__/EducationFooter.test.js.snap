// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EducationFooter /> should render and match the snapshot 1`] = `
<div
  className="d-flex flex-column align-items-center flex-md-row justify-content-md-end my-5"
>
  <ForwardRef(LinkButtonMod__ButtonMod)
    color="link"
    onClick={[Function]}
  >
    <FormattedMessage
      defaultMessage="Skip"
      id="CodeMonk.containers.containerMessage.skipButton"
      values={Object {}}
    />
  </ForwardRef(LinkButtonMod__ButtonMod)>
  <Button
    className="btn-primary mt-3 mt-md-0 ms-md-3"
    disabled={true}
    onClick={[Function]}
  >
    <FormattedMessage
      defaultMessage="Continue"
      id="CodeMonk.containers.containerMessage.continueButton"
      values={Object {}}
    />
  </Button>
</div>
`;
